nat-lab/tests/test_network_monitor.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+import time
+from contextlib import AsyncExitStack
+from helpers import setup_mesh_nodes, SetupParameters
+from telio import AdapterType
+from utils.connection_util import ConnectionTag
+
+DEFAULT_WAITING_TIME = 2
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "alpha_setup_params",
+    [
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.DOCKER_SHARED_CLIENT_1,
+                adapter_type=AdapterType.BoringTun,
+            ),
+        ),
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.WINDOWS_VM_1,
+                adapter_type=AdapterType.WireguardGo,
+            ),
+            marks=[pytest.mark.windows],
+        ),
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.MAC_VM,
+                adapter_type=AdapterType.BoringTun,
+            ),
+            marks=[
+                pytest.mark.mac,
+            ],
+        ),
+    ],
+)
+async def test_network_monitor(
+    alpha_setup_params: SetupParameters,
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        env = await setup_mesh_nodes(exit_stack, [alpha_setup_params])
+        [client_alpha] = env.clients
+
+        time.sleep(DEFAULT_WAITING_TIME)
```

Comment:
async sleep perhaps?

Reply:
why do we have to wait? try to avoid time dependency on tests, it increases flakyness probability